### ISSUE TITLE
No ability to inject own scheduler

### ISSUE THREAD

**Issue by ivanmkc** [sunshinejr/Moya-ModelMapper#51]
I'm using RxTest to test my network services which makes use of a virtual time TestScheduler. However, in Single+ModelMapper, you have fixed schedulers which are incompatible with RxTest as they would have to be replaced by the TestScheduler for unit tests. Similar discussion is here: [https://github.com/ReactiveX/RxSwift/issues/1100](https://github.com/ReactiveX/RxSwift/issues/1100)

Fix would be to do dependency injection of schedulers, perhaps with a default value.

Problematic code here:
```swift
public func map<T: Mappable>(to type: T.Type, keyPath: String? = nil) -> Single<T> {
        return observeOn(ConcurrentDispatchQueueScheduler(qos: .background))
            .flatMap { response -> Single<T> in
                return Single.just(try response.map(to: type, keyPath: keyPath))
            }
            .observeOn(MainScheduler.instance)
    }
````

**Comment by sunshinejr** [comment]
Hi @ivanmkc. You are right, it is a bug on my end. [I removed the `observeOn` lines from `Observable` extension](https://github.com/sunshinejr/Moya-ModelMapper/commit/4af17ff44471429f6f6fa3a9ebb2ba3c6e789918), but forgot to do it for `Single`. Would you be up for removing these in a pull request? I'd be more than happy to review it.

**Comment by sunshinejr** [comment]
@ivanmkc Yeah, I think that developers should handle switching between schedulers themselves (your alternative solution).

**Comment by ivanmkc** [comment]
Ok, I tried to search for some RxSwift guideline on where it states this what couldn't find anything. I will submit a PR with the alternative. 

Thanks!

**Comment by ivanmkc** [comment]
Submitted a PR.

**Comment by sunshinejr** [comment]
@ivanmkc there is no strict rule for managing schedulers in external libraries (as far as I know), so this is probably left out to developers.
